In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
DATA_PATH = "../data/raw/"

In [3]:
os.listdir(DATA_PATH)

['Plant_1_Generation_Data.csv',
 'Plant_1_Weather_Sensor_Data.csv',
 'Plant_2_Generation_Data.csv',
 'Plant_2_Weather_Sensor_Data.csv']

In [4]:
plant1_generation = pd.read_csv(
    DATA_PATH + "Plant_1_Generation_Data.csv"
)

plant1_weather = pd.read_csv(
    DATA_PATH + "Plant_1_Weather_Sensor_Data.csv"
)

plant2_generation = pd.read_csv(
    DATA_PATH + "Plant_2_Generation_Data.csv"
)

plant2_weather = pd.read_csv(
    DATA_PATH + "Plant_2_Weather_Sensor_Data.csv"
)

In [5]:
print("Plant 1 Generation:", plant1_generation.shape)
print("Plant 1 Weather:", plant1_weather.shape)

print("Plant 2 Generation:", plant2_generation.shape)
print("Plant 2 Weather:", plant2_weather.shape)

Plant 1 Generation: (68778, 7)
Plant 1 Weather: (3182, 6)
Plant 2 Generation: (67698, 7)
Plant 2 Weather: (3259, 6)


In [6]:
print("Plant 1 Generation Columns:")
print(plant1_generation.columns.tolist())

print("\nPlant 1 Weather Columns:")
print(plant1_weather.columns.tolist())

print("\nPlant 2 Generation Columns:")
print(plant2_generation.columns.tolist())

print("\nPlant 2 Weather Columns:")
print(plant2_weather.columns.tolist())

Plant 1 Generation Columns:
['DATE_TIME', 'PLANT_ID', 'SOURCE_KEY', 'DC_POWER', 'AC_POWER', 'DAILY_YIELD', 'TOTAL_YIELD']

Plant 1 Weather Columns:
['DATE_TIME', 'PLANT_ID', 'SOURCE_KEY', 'AMBIENT_TEMPERATURE', 'MODULE_TEMPERATURE', 'IRRADIATION']

Plant 2 Generation Columns:
['DATE_TIME', 'PLANT_ID', 'SOURCE_KEY', 'DC_POWER', 'AC_POWER', 'DAILY_YIELD', 'TOTAL_YIELD']

Plant 2 Weather Columns:
['DATE_TIME', 'PLANT_ID', 'SOURCE_KEY', 'AMBIENT_TEMPERATURE', 'MODULE_TEMPERATURE', 'IRRADIATION']


In [7]:
plant1_generation.head()


,DATE_TIME,PLANT_ID,SOURCE_KEY,DC_POWER,AC_POWER,DAILY_YIELD,TOTAL_YIELD
0,15-05-2020 00:00,4135001,1BY6WEcLGh8j5v7,0.0,0.0,0.0,6259559.0
1,15-05-2020 00:00,4135001,1IF53ai7Xc0U56Y,0.0,0.0,0.0,6183645.0
2,15-05-2020 00:00,4135001,3PZuoBAID5Wc2HD,0.0,0.0,0.0,6987759.0
3,15-05-2020 00:00,4135001,7JYdWkrLSPkdwr4,0.0,0.0,0.0,7602960.0
4,15-05-2020 00:00,4135001,McdE0feGgRqW7Ca,0.0,0.0,0.0,7158964.0


In [8]:
plant1_weather.head()

,DATE_TIME,PLANT_ID,SOURCE_KEY,AMBIENT_TEMPERATURE,MODULE_TEMPERATURE,IRRADIATION
0,2020-05-15 00:00:00,4135001,HmiyD2TTLFNqkNe,25.184316,22.857507,0.0
1,2020-05-15 00:15:00,4135001,HmiyD2TTLFNqkNe,25.084589,22.761668,0.0
2,2020-05-15 00:30:00,4135001,HmiyD2TTLFNqkNe,24.935753,22.592306,0.0
3,2020-05-15 00:45:00,4135001,HmiyD2TTLFNqkNe,24.846130,22.360852,0.0
4,2020-05-15 01:00:00,4135001,HmiyD2TTLFNqkNe,24.621525,22.165423,0.0


In [10]:
plant2_generation.head()


,DATE_TIME,PLANT_ID,SOURCE_KEY,DC_POWER,AC_POWER,DAILY_YIELD,TOTAL_YIELD
0,2020-05-15 00:00:00,4136001,4UPUqMRk7TRMgml,0.0,0.0,9425.000000,2.429011e+06
1,2020-05-15 00:00:00,4136001,81aHJ1q11NBPMrL,0.0,0.0,0.000000,1.215279e+09
2,2020-05-15 00:00:00,4136001,9kRcWv60rDACzjR,0.0,0.0,3075.333333,2.247720e+09
3,2020-05-15 00:00:00,4136001,Et9kgGMDl729KT4,0.0,0.0,269.933333,1.704250e+06
4,2020-05-15 00:00:00,4136001,IQ2d7wF4YD8zU1Q,0.0,0.0,3177.000000,1.994153e+07


In [11]:
plant2_weather.head()

,DATE_TIME,PLANT_ID,SOURCE_KEY,AMBIENT_TEMPERATURE,MODULE_TEMPERATURE,IRRADIATION
0,2020-05-15 00:00:00,4136001,iq8k7ZNt4Mwm3w0,27.004764,25.060789,0.0
1,2020-05-15 00:15:00,4136001,iq8k7ZNt4Mwm3w0,26.880811,24.421869,0.0
2,2020-05-15 00:30:00,4136001,iq8k7ZNt4Mwm3w0,26.682055,24.427290,0.0
3,2020-05-15 00:45:00,4136001,iq8k7ZNt4Mwm3w0,26.500589,24.420678,0.0
4,2020-05-15 01:00:00,4136001,iq8k7ZNt4Mwm3w0,26.596148,25.088210,0.0


In [12]:
print("Plant 1 Generation:")
plant1_generation.info()

print("\nPlant 1 Weather:")
plant1_weather.info()

print("\nPlant 2 Generation:")
plant2_generation.info()

print("\nPlant 2 Weather:")
plant2_weather.info()

Plant 1 Generation:
<class 'pandas.DataFrame'>
RangeIndex: 68778 entries, 0 to 68777
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   DATE_TIME    68778 non-null  str    
 1   PLANT_ID     68778 non-null  int64  
 2   SOURCE_KEY   68778 non-null  str    
 3   DC_POWER     68778 non-null  float64
 4   AC_POWER     68778 non-null  float64
 5   DAILY_YIELD  68778 non-null  float64
 6   TOTAL_YIELD  68778 non-null  float64
dtypes: float64(4), int64(1), str(2)
memory usage: 5.7 MB

Plant 1 Weather:
<class 'pandas.DataFrame'>
RangeIndex: 3182 entries, 0 to 3181
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   DATE_TIME            3182 non-null   str    
 1   PLANT_ID             3182 non-null   int64  
 2   SOURCE_KEY           3182 non-null   str    
 3   AMBIENT_TEMPERATURE  3182 non-null   float64
 4   MODULE_TEMPERATURE   3182 non

In [13]:
print("Plant 1 Generation Missing Values:")
print(plant1_generation.isnull().sum())

print("\nPlant 1 Weather Missing Values:")
print(plant1_weather.isnull().sum())

print("\nPlant 2 Generation Missing Values:")
print(plant2_generation.isnull().sum())

print("\nPlant 2 Weather Missing Values:")
print(plant2_weather.isnull().sum())

Plant 1 Generation Missing Values:
DATE_TIME      0
PLANT_ID       0
SOURCE_KEY     0
DC_POWER       0
AC_POWER       0
DAILY_YIELD    0
TOTAL_YIELD    0
dtype: int64

Plant 1 Weather Missing Values:
DATE_TIME              0
PLANT_ID               0
SOURCE_KEY             0
AMBIENT_TEMPERATURE    0
MODULE_TEMPERATURE     0
IRRADIATION            0
dtype: int64

Plant 2 Generation Missing Values:
DATE_TIME      0
PLANT_ID       0
SOURCE_KEY     0
DC_POWER       0
AC_POWER       0
DAILY_YIELD    0
TOTAL_YIELD    0
dtype: int64

Plant 2 Weather Missing Values:
DATE_TIME              0
PLANT_ID               0
SOURCE_KEY             0
AMBIENT_TEMPERATURE    0
MODULE_TEMPERATURE     0
IRRADIATION            0
dtype: int64


In [14]:
print("Plant 1 Generation duplicates:",
      plant1_generation.duplicated().sum())

print("Plant 1 Weather duplicates:",
      plant1_weather.duplicated().sum())

print("Plant 2 Generation duplicates:",
      plant2_generation.duplicated().sum())

print("Plant 2 Weather duplicates:",
      plant2_weather.duplicated().sum())

Plant 1 Generation duplicates: 0
Plant 1 Weather duplicates: 0
Plant 2 Generation duplicates: 0
Plant 2 Weather duplicates: 0


In [15]:
for name, df in {
    "Plant 1 Generation": plant1_generation,
    "Plant 1 Weather": plant1_weather,
    "Plant 2 Generation": plant2_generation,
    "Plant 2 Weather": plant2_weather
}.items():
    
    print(name)
    print("Start:", df["DATE_TIME"].min())
    print("End  :", df["DATE_TIME"].max())
    print()

Plant 1 Generation
Start: 01-06-2020 00:00
End  : 31-05-2020 23:45

Plant 1 Weather
Start: 2020-05-15 00:00:00
End  : 2020-06-17 23:45:00

Plant 2 Generation
Start: 2020-05-15 00:00:00
End  : 2020-06-17 23:45:00

Plant 2 Weather
Start: 2020-05-15 00:00:00
End  : 2020-06-17 23:45:00



In [16]:
# Step 12 — Check plants and inverters
print("Plant 1 Generation - Plants:")
print(plant1_generation["PLANT_ID"].unique())

print("\nPlant 2 Generation - Plants:")
print(plant2_generation["PLANT_ID"].unique())

Plant 1 Generation - Plants:
[4135001]

Plant 2 Generation - Plants:
[4136001]


In [17]:
print(
    "Plant 1 Generation inverters:",
    plant1_generation["SOURCE_KEY"].nunique()
)

print(
    "Plant 2 Generation inverters:",
    plant2_generation["SOURCE_KEY"].nunique()
)

Plant 1 Generation inverters: 22
Plant 2 Generation inverters: 22


In [18]:
print(
    "Plant 1 Weather sources:",
    plant1_weather["SOURCE_KEY"].nunique()
)

print(
    "Plant 2 Weather sources:",
    plant2_weather["SOURCE_KEY"].nunique()
)

Plant 1 Weather sources: 1
Plant 2 Weather sources: 1


In [19]:
# check numerical statics
plant1_generation.describe()

,PLANT_ID,DC_POWER,AC_POWER,DAILY_YIELD,TOTAL_YIELD
count,68778.0,68778.000000,68778.000000,68778.000000,6.877800e+04
mean,4135001.0,3147.426211,307.802752,3295.968737,6.978712e+06
std,0.0,4036.457169,394.396439,3145.178309,4.162720e+05
min,4135001.0,0.000000,0.000000,0.000000,6.183645e+06
25%,4135001.0,0.000000,0.000000,0.000000,6.512003e+06
50%,4135001.0,429.000000,41.493750,2658.714286,7.146685e+06
75%,4135001.0,6366.964286,623.618750,6274.000000,7.268706e+06
max,4135001.0,14471.125000,1410.950000,9163.000000,7.846821e+06


In [20]:
plant2_generation.describe()

,PLANT_ID,DC_POWER,AC_POWER,DAILY_YIELD,TOTAL_YIELD
count,67698.0,67698.000000,67698.000000,67698.000000,6.769800e+04
mean,4136001.0,246.701961,241.277825,3294.890295,6.589448e+08
std,0.0,370.569597,362.112118,2919.448386,7.296678e+08
min,4136001.0,0.000000,0.000000,0.000000,0.000000e+00
25%,4136001.0,0.000000,0.000000,272.750000,1.996494e+07
50%,4136001.0,0.000000,0.000000,2911.000000,2.826276e+08
75%,4136001.0,446.591667,438.215000,5534.000000,1.348495e+09
max,4136001.0,1420.933333,1385.420000,9873.000000,2.247916e+09


In [21]:
plant1_weather.describe()

,PLANT_ID,AMBIENT_TEMPERATURE,MODULE_TEMPERATURE,IRRADIATION
count,3182.0,3182.000000,3182.000000,3182.000000
mean,4135001.0,25.531606,31.091015,0.228313
std,0.0,3.354856,12.261222,0.300836
min,4135001.0,20.398505,18.140415,0.000000
25%,4135001.0,22.705182,21.090553,0.000000
50%,4135001.0,24.613814,24.618060,0.024653
75%,4135001.0,27.920532,41.307840,0.449588
max,4135001.0,35.252486,65.545714,1.221652


In [22]:
plant2_weather.describe()

,PLANT_ID,AMBIENT_TEMPERATURE,MODULE_TEMPERATURE,IRRADIATION
count,3259.0,3259.000000,3259.000000,3259.000000
mean,4136001.0,28.069400,32.772408,0.232737
std,0.0,4.061556,11.344034,0.312693
min,4136001.0,20.942385,20.265123,0.000000
25%,4136001.0,24.602135,23.716881,0.000000
50%,4136001.0,26.981263,27.534606,0.019040
75%,4136001.0,31.056757,40.480653,0.438717
max,4136001.0,39.181638,66.635953,1.098766


| Dataset | Rows | Columns | Missing | Duplicates |
|---|---:|---:|---:|---:|
| Plant 1 Generation | 68,778 | 7 | 0 | 0 |
| Plant 1 Weather | 3,182 | 6 | 0 | 0 |
| Plant 2 Generation | 67,698 | 7 | 0 | 0 |
| Plant 2 Weather | 3,259 | 6 | 0 | 0 |



